achievements.lua

```diff
@@ -139,21 +286,27 @@ function achievements.initialize(gamedata, prevent_debug)
 	end
 	if gamedata.gameID == nil then
 		gamedata.gameID = string.gsub(metadata.bundleID, "^user%.%d+%.", "")
-		print("gameID not configured: defaulting to \"" .. gamedata.gameID .. "\"")
+		print('gameID not configured: defaulting to "' .. gamedata.gameID .. '"')
 	elseif type(gamedata.gameID) ~= "string" then
 		error("gameID must be a string", 2)
 	end
-	for _, field in ipairs{"name", "author", "description"} do
+	for _, field in ipairs{ "name", "author", "description", "version", "buildNumber", } do
 		if gamedata[field] == nil then
-			gamedata[field] = playdate.metadata[field]
-			print(field .. " not configured: defaulting to \"" .. gamedata[field] .. "\"")
+			if playdate.metadata[field] ~= nil then
+				gamedata[field] = playdate.metadata[field]
+				print(field .. ' not configured: defaulting to "' .. gamedata[field] .. '"')
+			else
+				print("WARNING: " .. field .. " not configured AND not present in pxinfo metadata")
+			end
 		elseif type(gamedata[field]) ~= "string" then
 			error(field .. " must be a string", 2)
 		end
 	end
 	gamedata.version = metadata.version
-	gamedata.libversion = achievements.version
+	gamedata.specversion = achievements.specversion
+	gamedata.libversion = achievements.libversion
```

Comment:
`libversion` may be slightly irrelevant if the floor is being left open to alternative implementations. Worth discussing.

Reply:
There might be things an implementation might want to save that others don't and vice-versa (there is room in the specs to add your own stuff after all).

Though now that I'm thinking of that, in order for that to work, the implementation should also be named or identified in some way.

Reply:
I've seen libraries place module information such as names and licensing inside the code before, which I've always thought was a little unnecessary but it isn't a bad pattern if identification is needed anyways.
That said, in my mind encouraging independent sub-specs rather than additions to the existing specification isn't entirely desirable, but I'll acknowledge that at least planning for it isn't a bad idea.

Probably needs some wider conversation. For the purposes of this pull request that can happen later.